scripts/main/album.js

```diff
@@ -1264,59 +1328,52 @@ album.isUploadable = function () {
 		return false;
 	}
 
-	// For special cases of no album / smart album / etc. we return true.
-	// It's only for regular non-matching albums that we return false.
-	if (album.json === null || !album.json.owner_name) {
-		return true;
-	}
-
-	return album.json.owner_name === lychee.username;
+	// TODO: Comparison of numeric user IDs (instead of names) should be more robust
+	return album.json !== null && album.json.owner_name === lychee.username;
```

Comment:
Hummmmm :thinking:

Reply:
What do you want to tell us?

Reply:
I don't know what he wants to say but I think it shouldn't make a difference? Surely the usernames must be unique, so why would that be less robust? We also currently don't transfer the numeric user IDs to the front end I believe, although that could be easily rectified if needed...

Reply:
You are right in this specific case it doesn't matter. I thought about the case when a user is renamed. The ID remains constant over time, but the user name not necessarily. But of course, here we always compare the current user name with the current name of the owner of the album.

It is more a matter of taste in this case. This is what IDs are meant for.